src/lib/coerce.js

```diff
@@ -99,16 +99,18 @@ exports.valObjects = {
         // TODO 'values shouldn't be in there (edge case: 'dash' in Scatter)
         otherOpts: ['dflt', 'noBlank', 'strict', 'arrayOk', 'values'],
         coerceFunction: function(v, propOut, dflt, opts) {
-            if(opts.strict === true && typeof v !== 'string') {
-                propOut.set(dflt);
-                return;
-            }
+            if(typeof v !== 'string') {
+                var okToCoerce = (
+                    typeof v === 'number' ||
+                    v instanceof Date ||
+                    typeof v === 'boolean'
+                );
```

Comment:
I'm OK with this, but actually, I'm coming around to your original position that we should only coerce numbers. My new argument against `Date` is that it will be altered if the plot is saved as JSON (plot.ly or otherwise) but also how likely is it that the default stringification is what the user wants for their date? And then `'boolean'` just starts to look out of place.